avakas/avakas.py

```diff
@@ -76,24 +77,97 @@ def read(cls):
         elif bump == 'minor':
-            self._version = self._version.next_minor()
+            bump_method = self._version.next_minor
         elif bump == 'major':
-            self._version = self._version.next_major()
+            bump_method = self._version.next_major
         else:
             raise AvakasError("Invalid version component")
 
+        self._version = bump_method()
+
+        if prerelease:
+            prerelease_version = self._get_prerelease_version(
+                original, prefix=prerelease_prefix
+            )
+
+            self.apply_prerelease(prerelease_version,
+                                  prefix=prerelease_prefix,
+                                  build_date=build_date)
+
+        assert self._version != original
+
         return True
 
+    def _get_prerelease_version(self, starting_version, prefix,
+                                new_version=None):
+        """
+        Return an integer representing the version of a given prerelase label
+        (i.e. alpha, beta, rc) which comes next
+        """
+
+        if new_version is None:
+            new_version = self._version
+
+        # This \/ checks whether last version was a pre-release, and then
+        # whether the beginning (at minimum) of the current pre-release
+        # prefix (PRP) matches the intended PRP. This could resolve to
+        # being `True` in the case that we're doing some sort of pre-pre
+        # release, e.g. rc.1.dev.1 would match an attempted 'rc' bump), but
+        # that's actually desirable (because bumping to the next 'rc'
+        # should treat `dev.1` as if it doesn't exist.
+        prerelease_len = 0
+        if not prefix:
+            prefix = tuple()
+        if isinstance(prefix, str):
+            prefix = (prefix, )
+
+        prerelease_len = len(prefix)
+
+        current_prefix_match = starting_version.prerelease[:prerelease_len]
+
+        if (new_version == starting_version.truncate() and
+                prefix == current_prefix_match):
+
+            # prerelease bumping for the same release.
+            # this will catch a case where there's e.g. rc.dev.1
+            # and we're trying to bump the 'rc' prefix, in that
+            # case per semver, there is an implicit zero (i.e. .alpha comes
+            # before .alpha.1).
+            try:
+                prerelease_version = int(
+                    starting_version.prerelease[prerelease_len])
+
+            except (IndexError, TypeError):
+                # either there is no explicit prerelease version in the
+                # current version, or there are additional prerelease
+                # prefixes which are not intended for this bump
+                prerelease_version = 0
+        else:
+            # different release or different prefix
+            prerelease_version = 0
+
+        prerelease_version += 1
+
+        return prerelease_version
+
     def make_prerelease(self, prefix=None, build_date=None):
```

Comment:
The `make_prerelease` method is now unused. Maybe that should be removed from the code? Not sure if it's intended on being a part of the 'public' Avakas API.

Reply:
oh yeah, I don't know if we know who depends on this, if we're being super strict removing `make_prerelease()` should happen on a major release, but it is a weekend and may be overindulging my foolish consistencies